### ISSUE TITLE
How to use with 3rd party endpoints?

### ISSUE THREAD

**Issue by vvakame** [google/google-api-go-client#109]
I'm using google-api-go-generator with own API of CloudEndpoints https://github.com/GoogleCloudPlatform/go-endpoints
just likes `google-api-go-generator -cache=false -discoveryurl=https://${discoveryHost}/_ah/api/discovery/v1/apis -api=user:v2 -gendir=${cwd}/src/foobar`

I have 2 issues now.

1. apiPackageBase is not configurable.
https://github.com/google/google-api-go-client/blob/master/google-api-go-generator/gen.go#L45
Custom Import Path Checking blocks compiling about our generated code.
by https://github.com/google/google-api-go-client/commit/09c6521081a8b11e2c277b747b24324cfdfa5c25

2. google-api-go-client uses internal dir.
our generated code can't access to google.golang.org/api/internal .
This is not problem just now. We use google appengine that sdk still go 1.4 base.

We need code generator for CloudEndpoints. :smiley:

**Comment by laco0416** [comment]
:+1: 
My project have the same problem. We want a independent repository for the generator!

**Comment by gmlewis** [comment]
Would making apiPackageBase a `var` solve the first problem?

For the second problem, what part(s) of `google.golang.org/api/internal` do you need to use?

**Comment by vvakame** [comment]
1. yes! I hope apiPackageBase switch to the command line option.
2. it is used by generated code. e.g. https://github.com/google/google-api-go-client/blob/master/taskqueue/v1beta2/taskqueue-gen.go#L20

**Comment by gmlewis** [comment]
My first attempt at a solution was apparently not correct.  We need to discuss the proper solution before attempting a fix.  /cc @bradfitz

**Comment by dsymonds** [comment]
The generator currently has a flag `-googleapi_pkg` to control the import path to use for the support package. However, the `internal` package seems to be another "support" package of sorts, but the generator does not permit controlling its import path, so the current state of affairs is now broken if you try to transplant all this into a different package hierarchy (e.g. vendoring it).

**Comment by gmlewis** [comment]
Dave, can you please look at https://code-review.googlesource.com/#/c/3540/ and let me know if you think it solves the problem?

**Comment by gmlewis** [comment]
Oh, I see, the source is already written and the comment line with the canonical import is already bad.
Ideas?

**Comment by dsymonds** [comment]
I don't see the problem with that CL for my particular use case. We can strip off the package import paths easily enough.

**Comment by gmlewis** [comment]
OK, thanks, Dave.  @bradfitz , can we please re-visit https://code-review.googlesource.com/#/c/3540/ ?

**Comment by bradfitz** [comment]
I don't know what anybody's requirements are, including @dsymonds's, so I'm not in a position to review any CL for suitability. I'll review once the CL description explains the problem and how it solves it in its commit message.

**Comment by gmlewis** [comment]
Sorry about that.  I've updated the CL description.
PTAL.

**Comment by gmlewis** [comment]
@vvakame and @laco0416, can you please perform an update and see if the flag change for `-api_pkg_base` in https://github.com/google/google-api-go-client/commit/bce17bfa8913f8777eb5221beb361c03edf5817e solves the problem for you?

If not, can you please elaborate on what the issue is and how you propose it should be fixed?

**Comment by vvakame** [comment]
great!
I tried it. and I found my truly requirement...

TL;DR. I need simply `-suppress_import_path_checking`.

In my project, We generate a code in CI environment.
CI process change target for code generation endpoints. staging or production.
Our project doest not have `/googleapi` and `/internal`. 
If we would copy it no longer the same as fork.

https://github.com/google/google-api-go-client/issues/109#issue-109721436
My issue No.1 is `Custom Import Path Checking blocks compiling about our generated code.`.
I have noticed that Custom Import Path Checking is not generated when We can compile generated code.
I need simply `-suppress_import_path_checking`.

**Comment by gmlewis** [comment]
Then if I understand correctly, this is no longer an issue.  Thanks for the update!
Closing.  Please re-open if there is still an issue.
